### PR TITLE
Update Custom Views docs

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Update Custom Views documentation.

#### Description

In this PR we try to improve the Custom Views docs around these topics:
* How to implement inner navigation in a Custom View
* How to fetch data from the current viewed entity in the host application (eg: product details)

### CODE REVIEW

**Review by kark — COMMENTED** [review]
Thank you 🙌

File: website-custom-views/src/content/development/navigation.mdx
Comment (emmenko) [inline website-custom-views/src/content/development/navigation.mdx:2]: I would suggest to name this `Routing`, as Navigation usually refers to the menu navigation.

File: website-custom-views/src/content/development/data-fetching.mdx
Comment (emmenko) [inline website-custom-views/src/content/development/data-fetching.mdx:698]: Wondering if we should provide an easier way to access the route params 🤔

**Review by emmenko — COMMENTED** [review]
Good job improving the docs, thanks!

File: website-custom-views/src/content/development/data-fetching.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/data-fetching.mdx:698]: Yes, that would be great although we have many different URLs in the Merchant Center so it would be difficult to maintain a clean strategy if we want to provide helpers like `getOrderIdFromUrl`, `getProductIdFromUrl`.

What do you have in mind?

File: website-custom-views/src/content/development/navigation.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/navigation.mdx]: Updated here: 9db9e6be40ffadb8edeec8b0c32a8a2962896c78

File: website-custom-views/src/content/development/data-fetching.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/data-fetching.mdx]: Updated here: 9db9e6be40ffadb8edeec8b0c32a8a2962896c78

File: website-custom-views/src/content/development/navigation.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/navigation.mdx]: Updated here: 9db9e6be40ffadb8edeec8b0c32a8a2962896c78

File: website-custom-views/src/content/development/navigation.mdx
Comment (CarlosCortizasCT) [reply website-custom-views/src/content/development/navigation.mdx]: Updated here: 9db9e6be40ffadb8edeec8b0c32a8a2962896c78

File: website-custom-views/src/content/development/data-fetching.mdx
Comment (Anshuman71) [inline website-custom-views/src/content/development/data-fetching.mdx:395]: I've removed the comments as they mess up with the VSCode config. We can always add back later.

**Review by Anshuman71 — APPROVED** [review]
Approving after the following change https://github.com/commercetools/merchant-center-application-kit/pull/3368/commits/34a156e666127e45d8bb6aee5272420772d8721e

**Comment by Anshuman71** [comment]
@CarlosCortizasCT let me know if my [changes](https://github.com/commercetools/merchant-center-application-kit/pull/3368) look good to you.

File: website-custom-views/src/content/development/data-fetching.mdx
Comment (emmenko) [reply website-custom-views/src/content/development/data-fetching.mdx]: I don't have anything specific in mind but I would keep it anyway as generic as possible.

In any case, we can also wait for feedback from users.

**Review by emmenko — APPROVED** [review]
💯

**Review by kark — APPROVED** [review]
🙌

**Comment by CarlosCortizasCT** [comment]
> @CarlosCortizasCT let me know if my [changes](https://github.com/commercetools/merchant-center-application-kit/pull/3368) look good to you.

Looks good to me 👍 

Thanks for your collaboration, @Anshuman71 🙇

**Pull request merged**
